doc/src/manual/variables-and-scoping.md

```diff
@@ -93,244 +102,356 @@ Note that the interactive prompt (aka REPL) is in the global scope of the module
+move code back and forth between a function body and the REPL when trying to debug the behavior of a
+function. However, it has some down sides. First, it's quite a complex behavior: many people over
+the years were confused about this behavior and complained that it was complicated and hard to
+understand and to explain. Fair point. Second, and arguably worse, is that it's bad for programming
+"at scale." When you see a small piece of code in one place like this, it's quite clear what's going
+on:
 
-julia> function quz()
-           x = 2 # local
-           return bar() + x # 12 + 2 (x is not modified)
-       end;
+```
+s = 0
+for i = 1:10
+    s += i
+end
+```
 
-julia> quz()
-14
+Obviously the intention is to modify the existing global variable `s`. What else could it mean?
+However, not all real world code is so short or so clear. We found that code like the following
+often occurs in the wild:
 
-julia> x, y # verify that global x and y are unchanged
-(1, 2)
 ```
+x = 123
 
-Note that the above nesting rules do not pertain to type and macro definitions as they can only appear
-at the global scope. There are special scoping rules concerning the evaluation of default and
-keyword function arguments which are described in the [Function section](@ref man-functions).
+# much later
+# maybe in a different file
 
-An assignment introducing a variable used inside a function, type or macro definition need not
-come before its inner usage:
+for i = 1:10
+    x = "hello"
+    println(x)
+end
 
-```jldoctest
-julia> f = y -> y + a;
+# much later
+# maybe in yet another file
+# or maybe back in the one where `x = 123` was
 
-julia> f(3)
-ERROR: UndefVarError: a not defined
-Stacktrace:
-[...]
+y = x + 234
+```
 
-julia> a = 1
-1
+Here it far less clear what should happen. Since `x + "hello"` is a method error, it seems probable
+that the intention is for `x` to be local to the `for` loop, but the types of runtime values and
+what methods happen to be defined cannot determine the scope of variables. With the Julia 0.6
+behavior, it's especially concerning that someone might have written the for loop first and it
+worked just fine, but then when someone adds a new global far away, possibly in a different file,
+the code suddenly changes meaning and either breaks noisily as in this case, or worse still, might
+break silently and just do the wrong thing. This kind of "spooky action at a distance" is the kind
+of thing good programming language designs should prevent.
+
+So in Julia 1.0, we simplified the rules for scope: in any local scope, assignment to a name that
+wasn't already a local variable created a new local variable. This eliminated the notion of soft
+scope entirely and eliminated the spooky action potential that having a global by a certain name
+previously had. We uncovered and fixed a log of bugs in the process. And there was much rejoicing!
+Well, no, not really. Because the people were angry that they now had to write:
```

Comment:
```suggestion
Well, no, not really. Because some people were angry that they now had to write:
```

Reply:
This choice of words was very intentional 😂